File: AD/codigos2425/T2/Agenda/Agenda/src/main/java/model/Usuario.java
package model;

import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import java.io.Serializable;

@Setter
@Getter
@AllArgsConstructor
@NoArgsConstructor
public class Usuario implements Serializable {
    private int id;
    private String nombre;
    private String correo;
    private String pass;
    private int telefono;
    private int idPerfil;

    public Usuario(String nombre, String correo, String pass, int telfono, int idPerfil) {
        this.nombre = nombre;
        this.correo = correo;
        this.pass = pass;
        this.telefono = telfono;
        this.idPerfil = idPerfil;
    }

    public void mostrarDatos(){
        System.out.println("id = " + id);
        System.out.println("nombre = " + nombre);
        System.out.println("correo = " + correo);
        System.out.println("pass = " + pass);
        System.out.println("telefono = " + telefono);
        System.out.println("idPerfil = " + idPerfil);
    }
}

File: PMDM/T1/InicioKT/src/main/kotlin/model/Usuario.kt
package model

open class Usuario(var nombre: String, var apellido: String, var dni: String) {

    // atributos nombre(o) apellido(o) dni(o) correo(?)
    var correo: String? = null
    var telefono: Int? = null

    init {
        // parte de codigo ejecutado siempre despues de un constructor
    }

    // constructores
    // primario --> base
    // secundarios --> base + adicional
    constructor(nombre: String, apellido: String, dni: String, correo: String) : this(nombre, apellido, dni) {
        this.correo = correo
    }

    constructor(nombre: String, apellido: String, dni: String, telefono: Int) : this(nombre, apellido, dni) {
        this.telefono = telefono
    }


    // metodos

    open fun mostrarDatos(): Unit {
        println("Los datos son: ")
        println("Nombre $nombre")
        println("Apellido $apellido")
        println("DNI $dni")
    }

    // getter-setter
}